### ISSUE TITLE
Watch History not being updated

### ISSUE THREAD

**Issue by Wolf68k** [yuliskov/SmartYouTubeTV#623]
Suddenly every video I watch is not being added to the watch history.

I have force quit the app, cleared the cache and (painfully) reinstalled the app.

How to recreate issue: Watch a video. Go check your history. It's not there.

App version: 6.17.702
Device: Amazon Fire TV Fire OS 5.2.7.3 (652613820)

**Comment by nierenquetsche** [comment]
Same problem here.

**Comment by Ph4te** [comment]
I can confirm this problem, it's the same on my Nvidia shield.
My YouTube account is saving its history if I'm using it on a normal browser. Smartyoutubetv is the only app to not save it. I already resetted it without any changes.

**Comment by shimmo10** [comment]
Yes same issue here.  Needs to be fixed in the next update

**Comment by jl410** [comment]
I have the same issue, no matter I use latest or beta version, it still can't save watch history.
but I use google original youtube app on the same box , it can save watch history.
Thanks for the developers, and hopefully it can be fixed on next release.

**Comment by nafarinha** [comment]
I'm also experiencing this issue

**Comment by liringea** [comment]
I also confirm this problem, this is due to the removal of ads, the same problem if you install any extension for Chrome (Windows) that removes ads, the browsing history also ceases to be saved. Smart youtube is installed on two TV boxes and have the same problem

**Comment by Wolf68k** [comment]
Ad blocking seems to be the reason. Going into Smart's settings and turning off ad blocking does allow the videos to be listed in the watch history.
Well that sucks.

Side note: When using Firefox with Nano Adblocker and it's blocking the ads and still the videos are being added to watch history.

**Comment by jl410** [comment]
I've checked my Chrome (Windows) extension uBlock origin extension to remove ads, and it seems OK to save watch history.
Then turn off uBlock origin and changed to using Remove Youtube Ads scripts (Chrome Wiondows), it also OK to save watch history.
It seems not all youtube ad block applications would cause this issue.

**Comment by yurividal** [comment]
Same here. Its pretty annoying. My home feed is just full of videos that I already watched.

**Comment by fguads** [comment]
Having the same issue using a 2019 shield TV pro, with any of the four of the app settings. App version 6.17.702

**Comment by batanglambak** [comment]
Same with me.

**Comment by chihche** [comment]
Same problem..

**Comment by Peter-Warlock** [comment]
The same problem. 
Here the log file. 
[log.txt.zip](https://github.com/yuliskov/SmartYouTubeTV/files/5089352/log.txt.zip)

**Comment by DARKFiB3R** [comment]
Same problem.

Interestingly (maybe), if I cast from Vanced to Smart, there no history save on either device.

**Comment by RedVisor** [comment]
Same problem as well; I've noticed that the problem only occurs on the PRO versions; the LITE versions are still saving the watch history.  Also, turning off ad blocking didn't make any difference for me.

That being said, I see that a new version (.709) was released around 10 minutes ago; my FireTV isn't picking up the update yet, but hopefully it will soon.

**Comment by foreveryoungforty** [comment]
To everyone : history is doing fine again with beta 710 just updated on my Sony 4K  😉👍

**Comment by nierenquetsche** [comment]
> 
> 
> To everyone : history is doing fine again with beta 710 just updated on my Sony 4K 😉👍

Thanks! with the beta it works again.

**Comment by iLLNESS** [comment]
> Same problem as well; I've noticed that the problem only occurs on the PRO versions; the LITE versions are still saving the watch history. Also, turning off ad blocking didn't make any difference for me.
> 
> That being said, I see that a new version (.709) was released around 10 minutes ago; my FireTV isn't picking up the update yet, but hopefully it will soon.

Can confirm, switching to Lite fixed it for me. Guess this will do until the update pops up for me.

**Comment by Wolf68k** [comment]
I've been on the LITE version this whole time and it's not working.

**Comment by RedVisor** [comment]
> Can confirm, switching to Lite fixed it for me. Guess this will do until the update pops up for me.

So far, yuliskov has only released the fix as a Beta update (.710).  If you're using the stable release apk then you'll need to download the beta release apk in order to use the PRO versions, at least until yuliskov commits it to a stable release.  You can get it from his home page: https://smartyoutubetv.github.io/

**Comment by hmm3** [comment]
Is beta using the pro main, crashing a lot for everyone?  Since the beta updates started, it's been crashing almost immediately after playing a video for a few seconds, or when fast forwarding.  Using Lite seems stop the crashes.  Turning on and off ad block doesn't seem to make a difference.  The release version seems to still work fine, although home page suggestions still aren't working or being updated according to your watch history.

**Comment by foreveryoungforty** [comment]
> Is beta using the pro main, crashing a lot for everyone? Since the beta updates started, it's been crashing almost immediately after playing a video for a few seconds, or when fast forwarding. Using Lite seems stop the crashes. Turning on and off ad block doesn't seem to make a difference. The release version seems to still work fine, although home page suggestions still aren't working or being updated according to your watch history.

As you know, the best launcher to use depends on capabilities of your device. Over here pro main always and still (beta 712) works great on my Sony 4K 65xf9005 .... no ads, no crashes, and history working fine again.

**Comment by hmm3** [comment]
> > Is beta using the pro main, crashing a lot for everyone? Since the beta updates started, it's been crashing almost immediately after playing a video for a few seconds, or when fast forwarding. Using Lite seems stop the crashes. Turning on and off ad block doesn't seem to make a difference. The release version seems to still work fine, although home page suggestions still aren't working or being updated according to your watch history.
> 
> As you know, the best launcher to use depends on capabilities of your device. Over here pro main always and still (beta 712) works great on my Sony 4K 65xf9005 .... no ads, no crashes, and history working fine again.

The latest updates seemed to fix the crashing problem.  Clicking on the ads or no ads block doesn't seem to do anything, either way I don't seem to get ads.  Is this because of I'm using adaway?  Although even when I turn that off I don't seem to get ads either.

**Comment by Wolf68k** [comment]
The latest stable release seems to fix for the Lite Main.

**Comment by nafarinha** [comment]
@Wolf68k the changelog for 6.17.715 says both LITE and PRO have the history fix. I did a small test and on my case, the history is working as expected for PRO as well.

**Comment by Wolf68k** [comment]
Yes I know. I was confirming what the history said when it tried to update.
Just because an history says something is fixed doesn't always make it so. I've seen many times when someone puts out an update and say something is fixed when it's really not and they, usually, soon release a hotfix update that actually does fix the thing they said was fixed before. 

Noticed I also I said 'tried to update'. While my FireTV did get the update when it tried to apply it, it came back saying that the update didn't install. So I went to the website and downloaded it from there to get the update. That was 715. Then 717 came out and that updated just fine.
